### PR TITLE
Basic Abacus Widget Design

### PR DESCRIPTION
**Pull request by DanielBeer**
https://trello.com/c/1HJQVOoX/24-basic-abacus-widget-design

flake8 completed

### CODE REVIEW

**Review by n0remac — APPROVED** [review]

File: wandering-warriors/main.py [hunk 1]
```diff
@@ -17,7 +18,136 @@ class TopMenu(Widget):
 
 
 class Abacus(Widget):
```
Comment (A5rocks) [inline wandering-warriors/main.py:20]: Could we move the whole class to another file?

File: wandering-warriors/main.py [hunk 2]
```diff
@@ -17,7 +18,136 @@ class TopMenu(Widget):
...
+
+        self.n_bars = 12
+        self.n_top_beads = 1
+        self.n_bottom_beads = 4
+        self.bar_w = self.MAX_BAR_W
+
+        with self.canvas:
+            Color(1, 1, 1, 1)
+
+            self.border = []
```
Comment (A5rocks) [inline wandering-warriors/main.py:36]: I don't feel this should be spaced out so much?

File: wandering-warriors/main.py [hunk 3]
```diff
@@ -17,7 +18,136 @@ class TopMenu(Widget):
...
+
+    def __init__(self, **kwargs):
+        super(Abacus, self).__init__(**kwargs)
+
+        self.n_bars = 12
+        self.n_top_beads = 1
+        self.n_bottom_beads = 4
+        self.bar_w = self.MAX_BAR_W
+
+        with self.canvas:
```
Comment (A5rocks) [inline wandering-warriors/main.py:33]: Maybe move this entire context block into a function called `create_background`?

File: wandering-warriors/main.py [hunk 4]
```diff
@@ -17,7 +18,136 @@ class TopMenu(Widget):
...
+                self.top_beads.append(top_beads)
+
+                bottom_beads = []
+
+                for j in range(self.n_bottom_beads):
+                    bottom_beads.append(Rectangle(source='assets/graphics/bead.png'))
+
+                self.bottom_beads.append(bottom_beads)
+
+        self.bind(pos=self.update, size=self.update)
```
Comment (A5rocks) [inline wandering-warriors/main.py:76]: Can't we do this in the `.kv` file?

**Review by A5rocks — APPROVED** [review]
Just some general refactoring: this is a *lot* of code to have in the main file.

**Pull request merged**